dig.go

```diff
@@ -128,6 +129,45 @@ func Group(group string) ProvideOption {
 	})
 }
 
+// ID is a unique integer representing the constructor node in the dependency graph.
+type ID int
+
+// ConstructorInfo provides information about the constructor's inputs and outputs
+// types as strings, as well as the ID of the constructor supplied to the Container.
+// It contains ID for the constructor, as well as slices of Input and Output types,
+// which are Stringers that report the types of the parameters and results respectively.
+type ConstructorInfo struct {
+	ID      ID
+	Inputs  []*Input
+	Outputs []*Output
+}
+
+// Input is a stringer that report the type of an input parameters of the constructor.
+type Input struct {
+	t reflect.Type
```

Comment:
Looks like we're losing some information:

- name of the input, if any
- whether the input is optional
- value group of the input, if any

The dot.Param has this information, FWIW.